client/app/components/dashboards/AddWidgetDialog.jsx

```diff
@@ -202,38 +206,27 @@ class AddWidgetDialog extends React.Component {
   }
 
   renderSearchQueryResults() {
+    const { isLoaded, queries, highlightSearchTerm, searchTerm } = this.state;
+
+    const highlightSearchResult = highlightSearchTerm ? highlight : identity;
+
     return (
       <div className="scrollbox" style={{ maxHeight: '50vh' }}>
-        {(this.state.searchTerm === '') && (
-          <div>
-            {this.state.recentQueries.length > 0 && (
-              <div className="list-group">
-                {this.state.recentQueries.map(query => (
-                  <a
-                    href="javascript:void(0)"
-                    className="list-group-item"
-                    key={query.id}
-                    onClick={() => this.selectQuery(query.id)}
-                  >
-                    {query.name}
-                    {' '}
-                    <QueryTagsControl tags={query.tags} className="inline-tags-control" />
-                  </a>
-                ))}
-              </div>
-            )}
+        {!isLoaded && (
```

Comment:
That was really a fragment that needed refactoring (99% duplicated code) - hey CodeClimate, where you was then?